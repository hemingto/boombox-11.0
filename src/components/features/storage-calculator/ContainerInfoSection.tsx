/**
 * @fileoverview Container information section for storage calculator
 * @source boombox-10.0/src/app/components/storagecalculator/containerinfosection.tsx
 * 
 * COMPONENT FUNCTIONALITY:
 * Displays key information about Boombox storage units including dimensions,
 * construction materials, and weatherproofing features. Includes an interactive
 * modal showing exact interior and exterior dimensions.
 * 
 * DESIGN SYSTEM UPDATES:
 * - Replaced bg-slate-100 placeholder with OptimizedImage component
 * - Updated to use semantic color tokens (text-text-secondary, bg-surface-tertiary)
 * - Replaced InformationalPopup with Modal component (per project preference)
 * - Applied design system typography and spacing patterns
 * 
 * @refactor 
 * - Renamed from containerinfosection.tsx to ContainerInfoSection.tsx (PascalCase)
 * - Replaced custom popup with Modal component from design system
 * - Added proper image optimization with OptimizedImage component
 * - Enhanced TypeScript interfaces for better type safety
 * - Improved accessibility with ARIA labels and semantic HTML
 */

'use client';

import React, { useState } from 'react';
import { RulerIcon, StorageUnitIcon, OpenStorageUnitIcon } from '@/components/icons';
import { Modal } from '@/components/ui/primitives/Modal/Modal';
import { OptimizedImage } from '@/components/ui/primitives/OptimizedImage/OptimizedImage';

interface FeatureProps {
  title: string;
  content: React.ReactNode;
  imageSrc?: string;
  imageAlt?: string;
}

interface DimensionModalContentProps {
  onClose: () => void;
}

/**
 * Modal content component displaying interior and exterior dimensions
 */
const DimensionModalContent: React.FC<DimensionModalContentProps> = ({ onClose }) => (
  <div className="p-6">
    <h2 className="text-2xl font-bold mb-4 text-text-primary">Exact Dimensions</h2>
    <p className="text-text-secondary mb-6">
      All measurements are as accurate as possible, but should be viewed as close approximations. 
      If you are unsure if one of your items will fit please reach out at our Help Center
    </p>
    
    <div className="flex flex-col sm:flex-row gap-8 mt-10">
      {/* Interior Dimensions */}
      <div className="flex-1 border-r border-border pr-6">
        <OpenStorageUnitIcon className="w-28 h-24 mb-4 mx-auto" aria-hidden="true" />
        <h3 className="mb-4 text-lg font-semibold text-center text-text-primary">
          Interior Dimensions
        </h3>
        <ul className="text-sm space-y-2 text-text-secondary" role="list">
          <li>
            <span className="font-semibold text-text-primary">Length:</span> 95 ins. or 7ft. 11 in. (241cm)
          </li>
          <li>
            <span className="font-semibold text-text-primary">Width:</span> 56 ins. or 4ft. 8 in. (142cm)
          </li>
          <li>
            <span className="font-semibold text-text-primary">Height:</span> 83.5 ins. or 6ft. 11.5 in. (212cm)
          </li>
          <li>
            <span className="font-semibold text-text-primary">Diagonal:</span> 110 ins. or 9ft. 2 in. (280cm)
          </li>
        </ul>
      </div>

      {/* Exterior Dimensions */}
      <div className="flex-1 pl-6">
        <StorageUnitIcon className="w-24 h-24 mb-4 mx-auto" aria-hidden="true" />
        <h3 className="mb-4 text-lg font-semibold text-center text-text-primary">
          Exterior Dimensions
        </h3>
        <ul className="text-sm space-y-2 text-text-secondary" role="list">
          <li>
            <span className="font-semibold text-text-primary">Length:</span> 96 ins. or 8ft. (244cm)
          </li>
          <li>
            <span className="font-semibold text-text-primary">Width:</span> 60 ins. or 5ft. (152cm)
          </li>
          <li>
            <span className="font-semibold text-text-primary">Height:</span> 83.5 ins. or 6ft. 11.5 in. (212cm)
          </li>
          <li>
            <span className="font-semibold text-text-primary">Diagonal:</span> 90 ins. or 7ft. 6 in. (228cm)
          </li>
        </ul>
      </div>
    </div>

    <div className="mt-8 flex justify-end">
      <button
        onClick={onClose}
        className="btn-primary"
        aria-label="Close dimensions modal"
      >
        Close
      </button>
    </div>
  </div>
);

/**
 * Individual feature card component
 */
const Feature: React.FC<FeatureProps> = ({ title, content, imageSrc, imageAlt }) => (
  <article className="flex flex-col mb-10">
    <div className="w-full aspect-square mb-4">
      {imageSrc ? (
        <OptimizedImage
          src={imageSrc}
          alt={imageAlt || title}
          width={400}
          height={400}
          aspectRatio="square"
          containerClassName="w-full rounded-md"
          className="object-cover rounded-md"
          loading="lazy"
          quality={85}
          sizes="(max-width: 768px) 100vw, (max-width: 1024px) 50vw, 33vw"
        />
      ) : (
        <OptimizedImage
          src="/placeholder.jpg"
          alt={`${title} illustration`}
          width={400}
          height={400}
          aspectRatio="square"
          containerClassName="w-full rounded-md"
          className="object-cover rounded-md"
          loading="lazy"
          quality={80}
          sizes="(max-width: 768px) 100vw, (max-width: 1024px) 50vw, 33vw"
        />
      )}
    </div>
    <h2 className="mb-2 text-xl font-semibold text-text-primary">{title}</h2>
    <div className="text-text-secondary">{content}</div>
  </article>
);

/**
 * Main container information section component
 */
export function ContainerInfoSection() {
  const [isDimensionsModalOpen, setIsDimensionsModalOpen] = useState(false);

  const features: FeatureProps[] = [
    {
      title: 'How we measure up',
      content: (
        <>
          <p className="mb-4">
            Boomboxes are 5 ft. wide 8ft. long and 8ft. tall. That&apos;s 320 cubic feet of storage, 
            enough to store a studio apartment full of items
          </p>
          <button
            onClick={() => setIsDimensionsModalOpen(true)}
            className="flex items-center cursor-pointer text-primary hover:text-primary-hover transition-colors"
            aria-label="View exact dimensions"
          >
            <RulerIcon className="w-6 mr-2" aria-hidden="true" />
            <span className="font-semibold underline">Exact Dimensions</span>
          </button>
        </>
      ),
    },
    {
      title: 'Sturdy steel construction',
      content: 'Boomboxes are constructed out of the highest grade steel to ensure your belongings are stored safely and securely',
    },
    {
      title: 'Weatherproof',
      content: 'Boomboxes are designed to hold up against the toughest of weather conditions and are water, wind, and dust proof',
    },
  ];

  return (
    <section 
      className="lg:px-16 px-6 sm:mb-48 mb-24"
      aria-labelledby="container-info-heading"
    >
      <h1 id="container-info-heading" className="mb-10 text-3xl font-bold text-text-primary">
        Learn more about your Boombox
      </h1>
      <div className="grid grid-cols-1 md:grid-cols-3 gap-4">
        {features.map((feature, index) => (
          <Feature key={index} {...feature} />
        ))}
      </div>

      {/* Dimensions Modal */}
      <Modal
        open={isDimensionsModalOpen}
        onClose={() => setIsDimensionsModalOpen(false)}
        size="lg"
        closeOnOverlayClick
      >
        <DimensionModalContent onClose={() => setIsDimensionsModalOpen(false)} />
      </Modal>
    </section>
  );
}

export default ContainerInfoSection;

