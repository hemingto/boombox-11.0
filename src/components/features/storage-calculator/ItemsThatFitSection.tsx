/**
 * @fileoverview Items that fit in a Boombox section
 * @source boombox-10.0/src/app/components/storagecalculator/itemsthatfitsection.tsx
 * 
 * COMPONENT FUNCTIONALITY:
 * Displays a horizontal scrollable list of large items that can fit in a Boombox
 * storage unit. Features smooth snap-to-item scrolling with navigation arrows.
 * 
 * DESIGN SYSTEM UPDATES:
 * - Replaced bg-slate-100 placeholder cards with OptimizedImage component
 * - Updated to use semantic color tokens (bg-surface-tertiary, text-text-primary)
 * - Applied design system button patterns for scroll controls
 * - Enhanced accessibility with proper ARIA labels and keyboard navigation
 * 
 * @refactor 
 * - Renamed from itemsthatfitsection.tsx to ItemsThatFitSection.tsx (PascalCase)
 * - Extracted scroll logic to useHorizontalScroll hook
 * - Replaced bg-slate placeholders with OptimizedImage component
 * - Improved TypeScript interfaces for better type safety
 * - Enhanced accessibility with semantic HTML and ARIA attributes
 */

'use client';

import React from 'react';
import Link from 'next/link';
import { ArrowLeftIcon, ArrowRightIcon } from '@heroicons/react/20/solid';
import { CheckIcon } from '@heroicons/react/20/solid';
import { useHorizontalScroll } from '@/hooks/useHorizontalScroll';
import { OptimizedImage } from '@/components/ui/primitives/OptimizedImage/OptimizedImage';

interface ItemCard {
  icon: React.ReactNode;
  subtitle: string;
  description: string;
  link: string;
  imageSrc?: string;
  imageAlt?: string;
}

/**
 * Main items that fit section component
 */
export function ItemsThatFitSection() {
  const { scrollContainerRef, handleScrollLeft, handleScrollRight } = useHorizontalScroll({
    gap: 16,
  });

  // Item content data
  const items: ItemCard[] = [
    {
      icon: <CheckIcon className="w-6 h-6" />,
      subtitle: 'King Mattress',
      description: 'A Boombox can fit a California King mattress standing up. The height of the container is 90" giving you about 6 inches of clearance for your mattress.',
      link: '/',
    },
    {
      icon: <CheckIcon className="w-6 h-6" />,
      subtitle: '3 Seat Sofa',
      description: 'The inside length of a Boombox is 90" and can fit most 3 seat sofas. Please make sure to measure before loading to make sure your sofa isn\'t longer than 90".',
      link: '/',
    },
    {
      icon: <CheckIcon className="w-6 h-6" />,
      subtitle: 'Large Dining Table',
      description: 'Your Boombox can fit most 8 seat dining tables. As long as the table top is less than 90". We recommend taking the legs off your dining table before loading to ensure it is safely stored.',
      link: '/',
    },
    {
      icon: <CheckIcon className="w-6 h-6" />,
      subtitle: '50 Medium Boxes',
      description: 'The Boombox can fit up to 50 medium boxes with the following dimensions 18-1/8" x 18" x 16". Make sure to stack lighter boxes on top of heavier boxes.',
      link: '/',
    },
  ];

  return (
    <section 
      className="sm:mb-48 mb-24"
      aria-labelledby="items-that-fit-heading"
    >
      {/* Header with navigation controls */}
      <div className="flex flex-col sm:flex-row w-full lg:px-16 px-6 justify-between items-left sm:items-center mb-10">
        <h1 
          id="items-that-fit-heading"
          className="text-left text-3xl font-bold text-text-primary"
        >
          What large items fit in a Boombox?
        </h1>
        <div className="flex mt-4 sm:mt-0 gap-2" role="group" aria-label="Scroll navigation">
          <button
            onClick={handleScrollLeft}
            className="rounded-full bg-surface-tertiary active:bg-surface-disabled hover:bg-surface-disabled transition-colors cursor-pointer p-2"
            aria-label="Scroll to previous item"
          >
            <ArrowLeftIcon className="w-6 text-text-primary" aria-hidden="true" />
          </button>
          <button
            onClick={handleScrollRight}
            className="rounded-full bg-surface-tertiary active:bg-surface-disabled hover:bg-surface-disabled transition-colors cursor-pointer p-2"
            aria-label="Scroll to next item"
          >
            <ArrowRightIcon className="w-6 text-text-primary" aria-hidden="true" />
          </button>
        </div>
      </div>

      {/* Scrollable items container */}
      <div
        id="scroll-container"
        className="w-full overflow-x-auto hide-scrollbar"
        ref={scrollContainerRef}
        tabIndex={0}
        role="region"
        aria-label="Scrollable items that fit in a Boombox"
      >
        <div
          id="item-container"
          className="lg:px-16 px-6 py-4 flex gap-4 flex-nowrap"
          role="list"
        >
          {items.map((item, index) => (
            <div key={index} role="listitem" className="flex-none">
              <Link 
                href={item.link}
              >
                <article className="bg-surface-tertiary w-[297.6px] sm:w-[372px] h-[569.6px] sm:h-[712px] rounded-md transform transition-transform duration-300 sm:hover:scale-[102%] cursor-pointer hover:z-10 relative overflow-hidden">
                {/* Image placeholder - could be replaced with actual product images */}
                <div className="absolute inset-0 -z-10">
                  <OptimizedImage
                    src={item.imageSrc || '/placeholder.jpg'}
                    alt={item.imageAlt || `${item.subtitle} in Boombox storage`}
                    width={372}
                    height={712}
                    aspectRatio="portrait"
                    containerClassName="w-full h-full"
                    className="object-cover opacity-10"
                    loading="lazy"
                    quality={80}
                    sizes="(max-width: 640px) 297.6px, 372px"
                  />
                </div>
                
                {/* Icon badge */}
                <div className="bg-surface-primary rounded-full py-2.5 px-2.5 font-semibold inline-block ml-4 mt-4 mb-2 text-sm">
                  <span className="text-status-success" aria-hidden="true">
                    {item.icon}
                  </span>
                </div>
                
                {/* Content */}
                <h2 className="ml-5 mb-2 text-xl font-semibold text-text-primary">
                  {item.subtitle}
                </h2>
                <p className="mx-5 text-text-secondary">
                  {item.description}
                </p>
              </article>
            </Link>
            </div>
          ))}
          
          {/* Spacer div for proper scrolling */}
          <div 
            className="bg-transparent lg:w-[48px] w-[8px] h-[569.6px] sm:h-[712px] flex-none" 
            aria-hidden="true"
          />
        </div>
      </div>

      {/* Hide scrollbar styles */}
      <style jsx>{`
        .hide-scrollbar::-webkit-scrollbar {
          display: none;
        }
        .hide-scrollbar {
          -ms-overflow-style: none;
          scrollbar-width: none;
        }
      `}</style>
    </section>
  );
}

export default ItemsThatFitSection;

