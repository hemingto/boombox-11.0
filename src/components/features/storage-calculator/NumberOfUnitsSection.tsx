/**
 * @fileoverview Number of units section showing Boombox requirements by home size
 * @source boombox-10.0/src/app/components/storagecalculator/numberofunitssection.tsx
 * 
 * COMPONENT FUNCTIONALITY:
 * Displays a table showing how many Boombox storage units are typically needed
 * for different home sizes (from extra items to full homes). Includes visual
 * icons representing 1, 2, or 3+ units.
 * 
 * DESIGN SYSTEM UPDATES:
 * - Replaced hardcoded colors (text-zinc-950, text-zinc-900) with semantic tokens
 * - Updated border colors from border-slate-100 to border-border
 * - Applied consistent heading styles with text-text-primary
 * - Enhanced accessibility with proper table structure and ARIA labels
 * 
 * @refactor 
 * - Renamed from numberofunitssection.tsx to NumberOfUnitsSection.tsx (PascalCase)
 * - Updated to use semantic color tokens from design system
 * - Improved TypeScript interfaces for better type safety
 * - Enhanced accessibility with semantic HTML and ARIA attributes
 * - Added proper table caption for screen readers
 */

import { StorageUnitIcon } from '@/components/icons/StorageUnitIcon';
import TwoUnitIcon from '@/components/icons/TwoUnitIcon';
import ThreeUnitIcon from '@/components/icons/ThreeUnitIcon';

interface StorageSize {
  size: string;
  containers: string;
  icon: React.FC<{ className?: string }>;
  iconWidth: string;
}

const storageData: StorageSize[] = [
  { 
    size: 'Extra Items', 
    containers: '1 unit', 
    icon: StorageUnitIcon, 
    iconWidth: 'w-12' 
  },
  { 
    size: 'Studio', 
    containers: '1-2 units', 
    icon: TwoUnitIcon, 
    iconWidth: 'w-16' 
  },
  { 
    size: '1 bedroom apartment', 
    containers: '2 units', 
    icon: TwoUnitIcon, 
    iconWidth: 'w-16' 
  },
  { 
    size: '2 bedroom apartment', 
    containers: '3 units', 
    icon: ThreeUnitIcon, 
    iconWidth: 'w-20' 
  },
  { 
    size: 'Full home', 
    containers: '3+ units', 
    icon: ThreeUnitIcon, 
    iconWidth: 'w-20' 
  },
];

/**
 * Main number of units section component
 */
export function NumberOfUnitsSection() {
  return (
    <section 
      className="lg:px-16 px-6 sm:mb-48 mb-24"
      aria-labelledby="units-by-size-heading"
    >
      <h1 
        id="units-by-size-heading"
        className="text-center mb-12 text-3xl font-bold text-text-primary"
      >
        Generally speaking
      </h1>
      
      <table className="w-full">
        <caption className="sr-only">
          Number of Boombox storage units needed by home size
        </caption>
        
        <thead className="border-b-2 border-border">
          <tr className="text-right">
            <th 
              scope="col" 
              className="py-2 text-left text-text-primary sr-only"
            >
              Home Size
            </th>
            <th 
              scope="col" 
              className="py-4 text-text-primary"
            >
              <h2 className="text-xl sm:text-2xl font-semibold">
                # of Boomboxes
              </h2>
            </th>
          </tr>
        </thead>
        
        <tbody className="border-b-2 border-border">
          {storageData.map((item, index) => {
            const IconComponent = item.icon;
            
            return (
              <tr key={index} className="text-text-secondary">
                <td className="py-6">
                  <h2 className="text-xl sm:text-2xl font-medium text-text-primary">
                    {item.size}
                  </h2>
                </td>
                <td className="py-6">
                  <div 
                    className="flex items-center space-x-4 justify-end"
                    role="group"
                    aria-label={`${item.containers} for ${item.size}`}
                  >
                    <span>
                      <h2 className="text-lg sm:text-xl font-medium text-text-primary">
                        {item.containers}
                      </h2>
                    </span>
                    <IconComponent 
                      className={`${item.iconWidth} text-text-primary`}
                      aria-hidden="true"
                    />
                  </div>
                </td>
              </tr>
            );
          })}
        </tbody>
      </table>
    </section>
  );
}

export default NumberOfUnitsSection;

