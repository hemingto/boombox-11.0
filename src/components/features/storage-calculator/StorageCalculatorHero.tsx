/**
 * @fileoverview Storage calculator hero section with heading and description
 * @source boombox-10.0/src/app/components/storagecalculator/storagecalculatorhero.tsx
 * 
 * COMPONENT FUNCTIONALITY:
 * Simple hero section for the storage calculator page. Displays an icon, heading,
 * and descriptive text to introduce the calculator functionality.
 * 
 * DESIGN SYSTEM UPDATES:
 * - Applied semantic text colors (text-text-primary, text-text-secondary)
 * - Used consistent spacing from design system (mt-12 sm:mt-24)
 * - Updated to use absolute imports with @/
 * - Enhanced accessibility with proper heading hierarchy
 * 
 * @refactor 
 * - Renamed from storagecalculatorhero.tsx to StorageCalculatorHero.tsx (PascalCase)
 * - Changed component name from StorageCalculatorHeroSection to StorageCalculatorHero for consistency
 * - Updated imports to use absolute paths
 * - Applied semantic design system color tokens
 * - Improved accessibility with semantic HTML structure
 */

'use client';

import { RulerIcon } from '@/components/icons/RulerIcon';

/**
 * Storage Calculator Hero Section Component
 * 
 * Displays the introductory section for the storage calculator with:
 * - Ruler icon for visual context
 * - Main heading "Storage Calculator"
 * - Descriptive subtitle
 */
export function StorageCalculatorHero() {
  return (
    <section 
      className="flex mt-12 sm:mt-24 lg:px-16 px-6"
      aria-labelledby="storage-calculator-heading"
    >
      <div className="place-content-center mb-8">
        <RulerIcon 
          className="mb-4 w-12 h-12 text-text-primary" 
          aria-hidden="true"
        />
        <h1 
          id="storage-calculator-heading"
          className="mb-4 text-text-primary"
        >
          Storage Calculator
        </h1>
        <p className="text-text-secondary">
          know exactly how much storage space you&apos;ll need
        </p>
      </div>
    </section>
  );
}

export default StorageCalculatorHero;

