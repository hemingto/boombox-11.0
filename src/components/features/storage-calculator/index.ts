/**
 * @fileoverview Storage calculator components
 * Exports all storage calculator related components
 */

export { ContainerInfoSection } from './ContainerInfoSection';
export { ItemsThatFitSection } from './ItemsThatFitSection';
export { NumberOfUnitsSection } from './NumberOfUnitsSection';
export { StorageCalculatorFAQ } from './StorageCalculatorFAQ';
export { StorageCalculatorHero } from './StorageCalculatorHero';
export { StorageCalculatorSection } from './StorageCalculatorSection';

