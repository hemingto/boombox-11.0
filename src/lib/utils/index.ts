/**
 * @fileoverview Central utility exports for boombox-11.0
 * Organized by domain and functionality for clean imports
 */

// Core utility functions
export * from './cn';
export * from './phoneUtils';
export * from './currencyUtils';
export * from './navigationUtils';
export * from './emailUtils';

// Notification utilities
export * from './notificationUtils';
export * from './validationUtils';
export * from './formatUtils';
export * from './statusUtils';

// Blog utilities
export * from './blogUtils';

// Explicit exports to resolve conflicts
export {
  calculateMonthlyStorageCost,
  isInServiceArea,
  getServiceAreaForZip,
} from './businessUtils';

export {
  formatDateForInput,
  formatDateForDisplay,
  formatDateCompact,
  formatDateTime,
  formatTime,
  formatTime24Hour,
  formatTimestamp,
  parseAppointmentTime,
  calculateDeliveryWindow,
  getUnitSpecificStartTime,
  formatVerboseDate,
} from './dateUtils';

export {
  generateDriverToken,
  generateCustomerMoverChangeToken,
  notifyDriverReassignment,
  notifyCustomerMoverChange,
  recordDriverCancellation,
  recordMoverCancellation,
  type DriverCancellationData,
  type MoverCancellationData,
  type AvailableDriver,
} from './cancellationUtils';

// Domain-specific utilities (avoiding conflicts)
// Note: appointmentUtils contains parseLoadingHelpPrice which conflicts with pricingUtils
// Using pricingUtils version for parseLoadingHelpPrice
export {
  formatAppointmentTime,
  formatAppointmentDateForSms,
  formatTimeMinusOneHour,
  validateAppointmentDateTime,
  calculateAppointmentChanges,
  calculateFinalUnitCount,
  getUnitNumbersToRemove,
  getDayOfWeekForAvailability,
  validateMovingPartnerAvailability,
  generateDriverReconfirmToken,
  generateDriverWebViewUrl,
  formatDriverInfo,
  verifyTrackingToken,
  // parseLoadingHelpPrice - using pricingUtils version
} from './appointmentUtils';

// Explicit exports from availabilityUtils to avoid conflicts
export {
  DEFAULT_BUSINESS_HOURS,
  calculateDriverRequirements,
  formatTimeLocal,
  generateBusinessHourSlots,
  hasTimeConflict,
  isResourceAvailableInSlot,
  createTimeConflict,
  checkOnfleetTaskConflicts,
  determineAvailabilityLevel,
  getDistinctDaysOfWeekInMonth,
  groupBy,
  isDateInPast,
  generateCacheKey,
  getMinutesDifference,
  isValidTimeSlot,
  timeStringToMinutes,
  // getDayOfWeekString - conflicts with dateFormattingUtils, using that version
} from './availabilityUtils';

export * from './moverChangeUtils';

// Explicit exports from movingPartnerUtils to avoid conflicts
export {
  createDefaultMoverAvailability,
  parseTimeToMinutes,
  checkMoverExists,
  createMover,
  findAvailableMovingPartners,
  uploadFileToCloudinary,
  deleteOldCloudinaryFile,
  getMovingPartnerJobs,
  assignMovingPartnerDriver,
  // getDayOfWeekString - conflicts with other modules, already exported from dateFormattingUtils
} from './movingPartnerUtils';

// Messaging and communication utilities
// Note: twilioUtils contains server-side code (headers) - import directly in API routes only
export * from './messageClassificationUtils';
export * from './inboundMessageUtils';
// Note: dateFormattingUtils contains formatAppointmentTime which conflicts with appointmentUtils
// Using appointmentUtils version for appointment-specific formatting
// export * from './dateFormattingUtils';
// Explicit export of non-conflicting functions from dateFormattingUtils
export { formatAppointmentDate } from './dateFormattingUtils';

// Driver utilities (explicit exports to avoid conflicts)
export {
  getDriverTeamIds,
  createDefaultDriverAvailability,
  formatPhoneForOnfleet,
  mapVehicleTypeToOnfleet,
  buildOnfleetWorkerPayload,
  validateDriverUniqueness,
  findDriverInvitation,
  validateInvitationStatus,
  DEFAULT_DRIVER_SERVICES,
  DEFAULT_MOVING_PARTNER_VEHICLE_TYPE,
  DEFAULT_DRIVER_CAPACITY,
  DRIVER_STATUSES,
  type DriverRegistrationData,
  type OnfleetWorkerPayload,
  type DriverApprovalResult,
} from './driverUtils';

// Pricing utilities (explicit exports to avoid conflicts)
export {
  getBoomboxPriceByZipCode,
  formatStorageUnitPrice,
  formatInsurancePrice,
  parseLoadingHelpPrice, // Using pricingUtils version as canonical
} from './pricingUtils';

// Storage utilities (explicit exports)
export {
  getStorageUnitText,
} from './storageUtils';

// Sorting and pagination utilities
export * from './sortingUtils';

// Tracking utilities
export * from './trackingStatusUtils';
