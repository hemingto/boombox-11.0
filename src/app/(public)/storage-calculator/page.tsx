/**
 * @fileoverview Storage calculator page - helps users estimate storage needs
 * @source boombox-10.0/src/app/storage-calculator/page.tsx
 * @refactor Migrated to (public) route group with SEO metadata
 */

'use client';

import type { Metadata } from 'next';
import {
  StorageCalculatorHero,
  StorageCalculatorSection,
  ItemsThatFitSection,
  NumberOfUnitsSection,
  ContainerInfoSection,
  StorageCalculatorFAQ,
} from '@/components/features/storage-calculator';
import { HowItWorksCustomerReviewSection } from '@/components/features/howitworks';
import { HelpCenterSection } from '@/components/features/landing';

// Note: Page is client component due to interactive calculator
// SEO metadata should be added via generateMetadata when converting to server component

export default function StorageCalculator() {
  return (
    <>
      <StorageCalculatorHero />
      <StorageCalculatorSection />
      <ItemsThatFitSection />
      <NumberOfUnitsSection />
      <ContainerInfoSection />
      <HowItWorksCustomerReviewSection />
      <StorageCalculatorFAQ />
      <HelpCenterSection />
    </>
  );
}

