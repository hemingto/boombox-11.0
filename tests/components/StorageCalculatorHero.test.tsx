/**
 * @fileoverview Tests for StorageCalculatorHero component
 * Following boombox-11.0 testing standards
 */

import { render, screen } from '@testing-library/react';
import { axe, toHaveNoViolations } from 'jest-axe';
import { StorageCalculatorHero } from '@/components/features/storage-calculator/StorageCalculatorHero';
import { testAccessibility } from '../utils/accessibility';

expect.extend(toHaveNoViolations);

// Mock RulerIcon component
jest.mock('@/components/icons/RulerIcon', () => ({
  RulerIcon: function MockRulerIcon({ className, ...props }: any) {
    return (
      <svg 
        data-testid="ruler-icon" 
        className={className}
        {...props}
      >
        <title>Ruler Icon</title>
      </svg>
    );
  }
}));

describe('StorageCalculatorHero', () => {
  // REQUIRED: Test cleanup
  beforeEach(() => {
    jest.clearAllMocks();
  });

  // ===========================
  // Rendering Tests
  // ===========================
  describe('Rendering', () => {
    it('renders without crashing', () => {
      render(<StorageCalculatorHero />);
      expect(screen.getByRole('region', { name: /storage calculator/i })).toBeInTheDocument();
    });

    it('renders the main heading', () => {
      render(<StorageCalculatorHero />);
      const heading = screen.getByRole('heading', { name: /storage calculator/i });
      expect(heading).toBeInTheDocument();
      expect(heading.tagName).toBe('H1');
    });

    it('renders the description text', () => {
      render(<StorageCalculatorHero />);
      expect(screen.getByText(/know exactly how much storage space you'll need/i)).toBeInTheDocument();
    });

    it('renders the ruler icon', () => {
      render(<StorageCalculatorHero />);
      expect(screen.getByTestId('ruler-icon')).toBeInTheDocument();
    });
  });

  // ===========================
  // Accessibility Tests (MANDATORY)
  // ===========================
  describe('Accessibility', () => {
    it('has no accessibility violations', async () => {
      const renderResult = render(<StorageCalculatorHero />);
      await testAccessibility(renderResult);
    });

    it('uses semantic HTML with proper heading hierarchy', () => {
      render(<StorageCalculatorHero />);
      const heading = screen.getByRole('heading', { name: /storage calculator/i });
      expect(heading.tagName).toBe('H1');
    });

    it('has proper ARIA landmark with labelledby', () => {
      render(<StorageCalculatorHero />);
      const section = screen.getByRole('region', { name: /storage calculator/i });
      expect(section).toHaveAttribute('aria-labelledby', 'storage-calculator-heading');
    });

    it('marks icon as decorative with aria-hidden', () => {
      render(<StorageCalculatorHero />);
      const icon = screen.getByTestId('ruler-icon');
      expect(icon).toHaveAttribute('aria-hidden', 'true');
    });

    it('has accessible text content with no empty elements', () => {
      const { container } = render(<StorageCalculatorHero />);
      
      // Verify heading has content
      const heading = screen.getByRole('heading', { name: /storage calculator/i });
      expect(heading.textContent).toBeTruthy();
      
      // Verify description has content
      const description = screen.getByText(/know exactly how much storage space you'll need/i);
      expect(description.textContent).toBeTruthy();
    });
  });

  // ===========================
  // Design System Integration Tests
  // ===========================
  describe('Design System Integration', () => {
    it('uses semantic color tokens for heading', () => {
      render(<StorageCalculatorHero />);
      const heading = screen.getByRole('heading', { name: /storage calculator/i });
      expect(heading).toHaveClass('text-text-primary');
    });

    it('uses semantic color tokens for description', () => {
      render(<StorageCalculatorHero />);
      const description = screen.getByText(/know exactly how much storage space you'll need/i);
      expect(description).toHaveClass('text-text-secondary');
    });

    it('uses semantic color tokens for icon', () => {
      render(<StorageCalculatorHero />);
      const icon = screen.getByTestId('ruler-icon');
      expect(icon).toHaveClass('text-text-primary');
    });

    it('does not use hardcoded color values', () => {
      const { container } = render(<StorageCalculatorHero />);
      const html = container.innerHTML;
      
      // Check for common hardcoded color patterns
      expect(html).not.toMatch(/text-zinc-\d+/);
      expect(html).not.toMatch(/text-gray-\d+/);
      expect(html).not.toMatch(/text-slate-\d+/);
      expect(html).not.toMatch(/bg-zinc-\d+/);
    });
  });

  // ===========================
  // Layout Structure Tests
  // ===========================
  describe('Layout Structure', () => {
    it('applies correct container spacing', () => {
      render(<StorageCalculatorHero />);
      const section = screen.getByRole('region', { name: /storage calculator/i });
      
      // Check responsive spacing classes
      expect(section).toHaveClass('mt-12');
      expect(section).toHaveClass('sm:mt-24');
      expect(section).toHaveClass('lg:px-16');
      expect(section).toHaveClass('px-6');
    });

    it('applies correct icon spacing', () => {
      render(<StorageCalculatorHero />);
      const icon = screen.getByTestId('ruler-icon');
      expect(icon).toHaveClass('mb-4');
      expect(icon).toHaveClass('w-12');
      expect(icon).toHaveClass('h-12');
    });

    it('applies correct heading spacing', () => {
      render(<StorageCalculatorHero />);
      const heading = screen.getByRole('heading', { name: /storage calculator/i });
      expect(heading).toHaveClass('mb-4');
    });

    it('uses proper flex layout', () => {
      render(<StorageCalculatorHero />);
      const section = screen.getByRole('region', { name: /storage calculator/i });
      expect(section).toHaveClass('flex');
    });
  });

  // ===========================
  // Content Tests
  // ===========================
  describe('Content', () => {
    it('displays correct heading text', () => {
      render(<StorageCalculatorHero />);
      expect(screen.getByRole('heading', { name: /storage calculator/i })).toHaveTextContent('Storage Calculator');
    });

    it('displays correct description with proper apostrophe', () => {
      render(<StorageCalculatorHero />);
      const description = screen.getByText(/know exactly how much storage space you'll need/i);
      expect(description).toBeInTheDocument();
      
      // Verify the apostrophe is properly escaped in the source
      expect(description.textContent).toContain("you'll");
    });

    it('has proper heading ID for ARIA relationship', () => {
      render(<StorageCalculatorHero />);
      const heading = screen.getByRole('heading', { name: /storage calculator/i });
      expect(heading).toHaveAttribute('id', 'storage-calculator-heading');
    });
  });

  // ===========================
  // Snapshot Test
  // ===========================
  describe('Snapshot', () => {
    it('matches snapshot', () => {
      const { container } = render(<StorageCalculatorHero />);
      expect(container).toMatchSnapshot();
    });
  });
});

