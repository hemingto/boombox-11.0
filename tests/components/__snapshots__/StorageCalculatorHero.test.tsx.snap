// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`StorageCalculatorHero Snapshot matches snapshot 1`] = `
<div>
  <section
    aria-labelledby="storage-calculator-heading"
    class="flex mt-12 sm:mt-24 lg:px-16 px-6"
  >
    <div
      class="place-content-center mb-8"
    >
      <svg
        aria-hidden="true"
        class="mb-4 w-12 h-12 text-text-primary"
        data-testid="ruler-icon"
      >
        <title>
          Ruler Icon
        </title>
      </svg>
      <h1
        class="mb-4 text-text-primary"
        id="storage-calculator-heading"
      >
        Storage Calculator
      </h1>
      <p
        class="text-text-secondary"
      >
        know exactly how much storage space you'll need
      </p>
    </div>
  </section>
</div>
`;
